contracts/staking/contracts/src/StakingProxy.sol

```diff
@@ -95,6 +97,44 @@ contract StakingProxy is
         emit ReadOnlyModeSet(readOnlyMode);
     }
 
+    /// @dev Batch executes a series of calls to the exchange contract.
+    /// @param data An array of data that encodes a sequence of functions to
+    ///             call in the staking contracts.
+    function batchExecute(bytes[] memory data)
+        public
+        returns (bytes[] memory batchReturnData)
+    {
+        // Initialize commonly used variables.
+        bool success;
+        bytes memory returnData;
+        batchReturnData = new bytes[](data.length);
+
+        // Ensure that a staking contract has been attached to the proxy.
+        if (stakingContract == address(0)) {
+            LibRichErrors.rrevert(
+                LibStakingRichErrors.ProxyDestinationCannotBeNilError()
+            );
+        }
+
+        // Execute all of the calls encoded in the provided calldata.
+        for (uint256 i = 0; i < data.length; i++) {
```

Comment:
Nit: Set `data.length` to a stack variable and use `!=` instead of `<`. This saves a tiny amount of gas, but it's primarily for consistency.